src/web/mod.rs

```diff
@@ -401,11 +397,13 @@ impl Server {
         conn: Arc<Mutex<Connection>>,
         config: Arc<Config>,
     ) -> Result<Self, Error> {
+        let templates = TemplateData::new(&conn.lock().unwrap())?;
```

Comment:
We should probably move away from using `Arc<Mutex>` since it means that any test that panics will cause other tests to panic. In the meantime though:
```suggestion
        let templates = TemplateData::new(&conn.lock().expect("failed to access shared database connection (did another test panic?)"))?;
```

Reply:
Hmm? Each test has its own instance of `TestDatabase`, with its own mutexed connection.

Reply:
Why do failures in one test cause others to fail then?

```
thread 'web::test::yanked_crates_are_not_considered' panicked at 'failed to lock the connection: "PoisonError { inner: .. }"', src/test/mod.rs:187:9
```

Reply:
Because we try to access the connection during `Drop`, and the Mutex will be poisoned if the test panicked.